pymtl3/passes/sim/SimpleTickPass.py

```diff
@@ -22,13 +23,44 @@ def iterative():
     return iterative
 
   def __call__( self, top ):
-    if not hasattr( top._sched, "schedule" ):
-      raise PassOrderError( "schedule" )
+    if not hasattr( top._sched, "update_schedule" ):
+      raise PassOrderError( "update_schedule" )
 
-    if hasattr( top, "_cl_trace" ):
-      schedule = top._cl_trace.schedule
-    else:
-      schedule = top._sched.schedule
+    # We assemble the final schedule from multiple sources of required
+    # work to generate a tick function for simulation
+
+    # Currently the tick order is:
+    # [ ff, tracing, posedge, clear_cl_trace, update ]
+
+    final_schedule = []
+    # call ff blocks first
+    final_schedule.extend( top._sched.schedule_ff )
+
+    # append tracing related work
 
+    if hasattr( top, "_tracing" ):
```

Comment:
What happens if both OpenLoopCLPass and VcdTracingPass are applied? Will the functions be inserted to _sched twice?

Reply:
No, the second VcdGenerationPass call will overwrite the previous function